dali/kernels/test/warp_test/warp_cpu_test.cc

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+#include <string>
+#include <vector>
+#include "dali/kernels/imgproc/warp_cpu.h"
+#include "dali/kernels/imgproc/warp/affine.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/dump_diff.h"
+#include "dali/kernels/test/mat2tensor.h"
+#include "dali/kernels/test/test_tensors.h"
+#include "dali/kernels/scratch.h"
+#include "dali/kernels/alloc.h"
+#include "dali/test/dali_test_config.h"
+#include "dali/core/geom/transform.h"
+#include "dali/kernels/test/warp_test/warp_test_helper.h"
+
+namespace dali {
+namespace kernels {
+
+inline void IsWarpKernelValid() {
+  check_kernel<WarpCPU<AffineMapping2D, 2, float, uint8_t, float>>();
+}
+
+TEST(WarpCPU, Affine_Transpose_Single) {
+  AffineMapping2D mapping_cpu = mat2x3{{
+    { 0, 1, 0 },
```

Comment:
Could you bring out the duplicated code for setting up these tests to fixture-class?
Then it would be possible to test it against different input-output pairs using `INPUT_OUTPUT_TYPED_TEST_SUITE` and code duplication will disappear

Reply:
I don't want to test Cartesian product of types here. I don't want to expose it in operator, either - as such, INPUT_OUTPUT_TYPED_TEST_SUITE is inadequate.

Reply:
Still you have bunch of duplicated code there. Plus, since this kernel should process variety of input-output type pair, there will be more (e.g. for `float16`, which test here also is lacking...)

Reply:
I don't see much duplicated code in this particular case, because it has a different validation procedure than the other ones.

Reply:
I tried to to it and it's definitely not shorter - it's a topic for another PR.